rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp

```diff
@@ -76,6 +77,88 @@ SqliteWrapper::~SqliteWrapper()
   }
 }
 
+void SqliteWrapper::apply_pragma_settings(
+  std::vector<std::string> & pragmas,
+  rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  // sqlite pragmas are assigned with either '= value' or '(value)' syntax, depending on pragma
+  const std::string pragma_assign = "=";
+  const std::string pragma_bracket = "(";
+
+  // Apply default pragmas if not overridden by user setting
+  {
+    // Used to check whether db is readable
+    const std::string schema = "schema_version";
+
+    typedef std::map<std::string, std::string> pragmas_map;
+    pragmas_map default_pragmas = {
+      {schema, ""}
+    };
+    if (io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE) {
+      const pragmas_map write_default_pragmas = {
+        {"journal_mode", pragma_assign + "WAL"},
+        {"synchronous", pragma_assign + "NORMAL"}
```

Comment:
I am not sure how we decided on the default values in the end, but that's the non-optimized pragmas, right? Have we hereby decided that these are the default values?

Reply:
If I recall correctly, we decided to change the defaults in a separate PR (along with changing the default split size to something else than 0 and including information about possible consequences)